### PR TITLE
Channelz: unifying OSS and internal

### PR DESCRIPTION
**Pull request by ncteisen**
First commit is re committing #16969 

Second commit refactors to unify build shims.

I will merge this only after submitting the internal change.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,014,439      Total (=)      2,014,439

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,113,676      Total (=)     11,113,676

 No significant differences in binary sizes


```

File: src/cpp/server/channelz/channelz_service.cc [hunk 1]
```diff
@@ -25,6 +25,135 @@
 
 namespace grpc {
 
+Status ChannelzService::GetTopChannels(
```
Comment (markdroth) [inline src/cpp/server/channelz/channelz_service.cc]: It looks like you've accidentally duplicated some code here.

Comment (ncteisen) [reply src/cpp/server/channelz/channelz_service.cc]: Ahh oops, was going to fast. Fixed

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,014,173      Total (=)      2,014,173

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,113,472      Total (<)     11,113,477

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Pull request merged**